cost-analyzer/templates/_helpers.tpl

```diff
@@ -31,6 +31,18 @@ Set important variables before starting main templates
   {{- end -}}
 {{- end -}}
 
+{{/*
+Kubecost 2.3 precondition that a full re-ingestion will be triggered for aggregator
+*/}}
+{{- define "kubecostV2.3-reingestion-precondition" -}}
+    {{- if semverCompare ">= 2.1.0-0 <2.4.0-0" .Chart.Version }}
+        {{- if gt .Release.Revision 1 -}}
+            {{- if not .Values.upgrade.acknowledgeV2dot3FullReingestion -}}
+                {{- fail "\n\nKubecost 2.3 requires a full reingestion of all data which will could take significant time. Please acknowledge by setting 'upgrade.acknowledgeV2dot3FullReingestion' to 'true' in your values file." -}}
```

Comment:
What if we have the value be upgrade.verifyVersion=2.3 and then if we need to use it in the future we use the same field and just increment the value?

Reply:
We can totally go this route. My concerns with that path are:
1) It is significantly less clear to the customer when they can remove the value from their values file
2) To someone looking at the file 6 months down the road - it is a lot harder to understand what this is, other than something they added in to make the errors go away. A more specific name - `lastAcknowledgedFullReingestionVersion=2.3` could help here.

Reply:
May want to consider the logic [here](https://github.com/kubecost/cost-analyzer-helm-chart/blob/13d4f5bcca8a302a7a2ee153b1b4af17ec6fc2ab/cost-analyzer/templates/_helpers.tpl#L37-L60) if you haven't already.